packages/aws-cdk/test/api/hotswap/hotswap-deployments.test.ts

```diff
@@ -0,0 +1,258 @@
+import { Lambda, StepFunctions } from 'aws-sdk';
+import * as setup from './hotswap-test-setup';
+
+let mockUpdateLambdaCode: (params: Lambda.Types.UpdateFunctionCodeRequest) => Lambda.Types.FunctionConfiguration;
+let mockUpdateMachineDefinition: (params: StepFunctions.Types.UpdateStateMachineInput) => StepFunctions.Types.UpdateStateMachineOutput;
+
+beforeEach(() => {
+  setup.setupHotswapTests();
+  mockUpdateLambdaCode = jest.fn();
+  mockUpdateMachineDefinition = jest.fn();
+  setup.setUpdateFunctionCodeMock(mockUpdateLambdaCode);
+  setup.setUpdateStateMachineMock(mockUpdateMachineDefinition);
+});
+
+test('returns a deployStackResult with noOp=true when it receives an empty set of changes', async () => {
+  // WHEN
+  const deployStackResult = await setup.tryHotswapDeployment({}, setup.cdkStackArtifactOf());
+
+  // THEN
+  expect(deployStackResult).not.toBeUndefined();
+  expect(deployStackResult?.noOp).toBeTruthy();
+  expect(deployStackResult?.stackArn).toEqual(setup.STACK_ID);
+});
+
+test('Resources that are not lambdas or step functions result in a full deployment', async () => {
+  // GIVEN
+  setup.setTemplate({
+    Resources: {
+      SomethingElse: {
+        Type: 'AWS::CloudFormation::SomethingElse',
+        Properties: {
+          Prop: 'old-value',
+        },
+      },
+    },
+  });
+  const cdkStackArtifact = setup.cdkStackArtifactOf({
+    template: {
+      Resources: {
+        SomethingElse: {
+          Type: 'AWS::CloudFormation::SomethingElse',
+          Properties: {
+            Prop: 'new-value',
+          },
+        },
+      },
+    },
+  });
+
+  // WHEN
+  const deployStackResult = await setup.tryHotswapDeployment({}, cdkStackArtifact);
+
+  // THEN
+  expect(deployStackResult).toBeUndefined();
+  expect(mockUpdateMachineDefinition).not.toHaveBeenCalled();
+  expect(mockUpdateLambdaCode).not.toHaveBeenCalled();
+});
+
+test('Resources that are not lambdas or step functions result in a full deployment even when a hotswappable change is present', async () => {
```

Comment:
It's not the _presence_ of the other resource that's a problem here, but a _change_ in a different resource.

I would change the name of the test to reflect that.